pkg/artifactory/resource_artifactory_remote_docker_repository.go

```diff
@@ -56,10 +56,21 @@ func resourceArtifactoryRemoteDockerRepository() *schema.Resource {
 			BlockPushingSchema1:          d.getBool("block_pushing_schema1", false),
 			ExternalDependenciesPatterns: d.getList("external_dependencies_patterns"),
 		}
+		if len(repo.ExternalDependenciesPatterns) == 0 {
```

Comment:
Actually, take a look at what @alexhung did in `resource_artifactory_ldap_settings`

```go
		"email_attribute": {
			Type:     schema.TypeString,
			Optional: true,
			Default:  "mail",
			StateFunc: func(str interface{}) string {
				// To match Artifactory behavior, Setting "email_attribute" value to "mail" when blank input is specified.
				value, ok := str.(string)
				if !ok || value == "" {
					return "mail"
				}
				return value
			},
			Description: `(Optional) An attribute that can be used to map a user's email address to a user created automatically in Artifactory. Default value is "mail".`,
		},
```

This is a more idiomatic approach